src/hooks/custom-session-provider.tsx

```diff
@@ -0,0 +1,43 @@
+import dayjs from 'dayjs';
+import { Session } from 'next-auth';
+import { SessionProvider, signIn } from 'next-auth/react';
+import { createContext, useContext, useEffect } from 'react';
+
+interface CustomSessionProviderProps {
+  session: Session;
+  children: React.ReactNode;
+  logoutUrl?: string;
+}
+
+export const CustomSessionProvider = ({
+  session,
+  children,
+  logoutUrl,
+}: CustomSessionProviderProps) => {
+  //Automatically redirect to the login page after a session expires or after 24 days
+  useEffect(() => {
+    if (!session) return () => {};
+    const timeout = setTimeout(signIn, Math.min(dayjs(session.expires).diff(), 2147483647));
```

Comment:
Store this number in a const and possible calculate it so it's more readable. Example: ``2 * 60 // 2 minutes``